core/src/test/scala/integration/kafka/api/BaseAsyncConsumerTest.scala

```diff
@@ -63,4 +69,156 @@ class BaseAsyncConsumerTest extends AbstractConsumerTest {
+    val consumer = createAsyncConsumer()
+    consumer.assign(List(tp).asJava)
+    // First consumer consumes and commits offsets
+    consumer.seek(tp, 0)
+    consumeAndVerifyRecords(consumer = consumer, numRecords, startingOffset = 0,
+      startingTimestamp = startingTimestamp)
+    consumer.commitSync()
+    assertEquals(numRecords, consumer.committed(Set(tp).asJava).get(tp).offset)
+    // We should see the committed offsets from another consumer
+    val anotherConsumer = createAsyncConsumer()
+    anotherConsumer.assign(List(tp).asJava)
+    assertEquals(numRecords, anotherConsumer.committed(Set(tp).asJava).get(tp).offset)
+  }
+
+  @Test
+  def testConsumeFromCommittedOffsets(): Unit = {
+    val producer = createProducer()
+    val numRecords = 100
+    val startingTimestamp = System.currentTimeMillis()
+    sendRecords(producer, numRecords = numRecords, tp, startingTimestamp = startingTimestamp)
+
+    // Commit offset with first consumer
+    val consumer = createConsumerWithGroupId("group1")
+    consumer.assign(List(tp).asJava)
+    val offset = 10
+    consumer.commitSync(Map[TopicPartition, OffsetAndMetadata]((tp, new OffsetAndMetadata(offset)))
+      .asJava)
+    assertEquals(offset, consumer.committed(Set(tp).asJava).get(tp).offset)
+    consumer.close()
+
+    // Consume from committed offsets with another consumer in same group
+    val anotherConsumer = createConsumerWithGroupId("group1")
+    assertEquals(offset, anotherConsumer.committed(Set(tp).asJava).get(tp).offset)
+    anotherConsumer.assign(List(tp).asJava)
+    consumeAndVerifyRecords(consumer = anotherConsumer, numRecords - offset,
+      startingOffset = offset, startingKeyAndValueIndex = offset,
+      startingTimestamp = startingTimestamp + offset)
+  }
+
+  @Test
+  def testRetrievingCommittedOffsetsMultipleTimes(): Unit = {
+    val numRecords = 100
+    val startingTimestamp = System.currentTimeMillis()
+    val producer = createProducer()
+    sendRecords(producer, numRecords, tp, startingTimestamp = startingTimestamp)
+
+    val consumer = createAsyncConsumer()
+    consumer.assign(List(tp).asJava)
+
+    // Consume and commit offsets
+    consumer.seek(tp, 0)
+    consumeAndVerifyRecords(consumer = consumer, numRecords, startingOffset = 0,
+      startingTimestamp = startingTimestamp)
+    consumer.commitSync()
+
+    // Check committed offsets twice with same consumer
+    assertEquals(numRecords, consumer.committed(Set(tp).asJava).get(tp).offset)
+    assertEquals(numRecords, consumer.committed(Set(tp).asJava).get(tp).offset)
+  }
+
+  @Test
+  def testEmptyGroupNotSupported(): Unit = {
+    // This replaces the existing testConsumingWithEmptyGroupId, given that empty group ID is not
+    // supported in the new consumer implementation
+    val consumer = createConsumerWithGroupId("")
+    consumer.assign(List(tp).asJava)
+    assertThrows(classOf[InvalidGroupIdException], () => consumer.commitSync())
```

Comment:
I assume that this change about the empty group id was done on purpose?

Reply:
Yes, I did it intentionally to align with the previous decision of not supporting empty group ID in the new consumer [KAFKA-14438](https://issues.apache.org/jira/browse/KAFKA-14438).